[Bug] [Dependent Task] error occured in dependent task node

### Search before asking

- [x] I had searched in the [issues](https://github.com/apache/dolphinscheduler/issues?q=is%3Aissue) and found no similar issues.


### What happened

```
2025-05-15 17:34:01.145 ERROR [shared-task-executor-container-worker-8] - LogicTaskExecutor{id=389, name=依赖, stat=RUNNING} execute failed
org.springframework.jdbc.BadSqlGrammarException: 
### Error querying database.  Cause: java.sql.SQLSyntaxErrorException: Unknown column 'instance.workflow_instance_id' in 'where clause'
### The error may exist in class path resource [org/apache/dolphinscheduler/dao/mapper/TaskInstanceMapper.xml]
### The error may involve defaultParameterMap
### The error occurred while setting parameters
### SQL: select                   instance.id, instance.name, instance.task_type, instance.task_code, instance.task_definition_version, instance.workflow_instance_id, instance.state, instance.submit_time,         instance.start_time, instance.end_time, instance.host, instance.execute_path, instance.log_path, instance.alert_flag, instance.retry_times, instance.pid, instance.app_link,         instance.flag, instance.retry_interval, instance.max_retry_times, instance.task_instance_priority, instance.worker_group,instance.environment_code , instance.executor_id,         instance.first_submit_time, instance.delay_time, instance.task_params, instance.var_pool, instance.dry_run, instance.test_flag, instance.task_group_id, instance.task_execute_type               from t_ds_task_instance instance         join (         select task_code, max(end_time) as max_end_time         from t_ds_task_instance         where 1=1 and test_flag = ?         and instance.workflow_instance_id = ?                       and task_code in              (                   ?              ,                  ?              )                    group by task_code         ) t_max         on instance.workflow_instance_id = t_max.workflow_instance_id         and instance.task_code = t_max.task_code         and instance.end_time = t_max.max_end_time
### Cause: java.sql.SQLSyntaxErrorException: Unknown column 'instance.workflow_instance_id' in 'where clause'
; bad SQL grammar []; nested exception is java.sql.SQLSyntaxErrorException: Unknown column 'instance.workflow_instance_id' in 'where clause'
	at org.springframework.jdbc.support.SQLErrorCodeSQLExceptionTranslator.doTranslate(SQLErrorCodeSQLExceptionTranslator.java:239)
	at org.springframework.jdbc.support.AbstractFallbackSQLExceptionTranslator.translate(AbstractFallbackSQLExceptionTranslator.java:70)
	at org.mybatis.spring.MyBatisExceptionTranslator.translateExceptionIfPossible(MyBatisExceptionTranslator.java:91)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:441)
	at com.sun.proxy.$Proxy128.selectList(Unknown Source)
	at org.mybatis.spring.SqlSessionTemplate.selectList(SqlSessionTemplate.java:224)
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.executeForMany(MybatisMapperMethod.java:166)
	at com.baomidou.mybatisplus.core.override.MybatisMapperMethod.execute(MybatisMapperMethod.java:77)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy$PlainMethodInvoker.invoke(MybatisMapperProxy.java:148)
	at com.baomidou.mybatisplus.core.override.MybatisMapperProxy.invoke(MybatisMapperProxy.java:89)
	at com.sun.proxy.$Proxy131.findLastTaskInstances(Unknown Source)
	at org.apache.dolphinscheduler.dao.repository.impl.TaskInstanceDaoImpl.queryLastTaskInstanceListIntervalInWorkflowInstance(TaskInstanceDaoImpl.java:171)
	at org.apache.dolphinscheduler.dao.repository.impl.TaskInstanceDaoImpl$$FastClassBySpringCGLIB$$5c344c57.invoke(<generated>)
	at org.springframework.cglib.proxy.MethodProxy.invoke(MethodProxy.java:218)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.invokeJoinpoint(CglibAopProxy.java:793)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:163)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.dao.support.PersistenceExceptionTranslationInterceptor.invoke(PersistenceExceptionTranslationInterceptor.java:137)
	at org.springframework.aop.framework.ReflectiveMethodInvocation.proceed(ReflectiveMethodInvocation.java:186)
	at org.springframework.aop.framework.CglibAopProxy$CglibMethodInvocation.proceed(CglibAopProxy.java:763)
	at org.springframework.aop.framework.CglibAopProxy$DynamicAdvisedInterceptor.intercept(CglibAopProxy.java:708)
	at org.apache.dolphinscheduler.dao.repository.impl.TaskInstanceDaoImpl$$EnhancerBySpringCGLIB$$53801737.queryLastTaskInstanceListIntervalInWorkflowInstance(<generated>)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.dependResultByAllTaskOfWorkflowInstance(DependentExecute.java:220)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.calculateResultForTasks(DependentExecute.java:167)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.getDependentResultForItem(DependentExecute.java:141)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.getDependResultForItem(DependentExecute.java:443)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.getModelDependResult(DependentExecute.java:416)
	at org.apache.dolphinscheduler.server.master.utils.DependentExecute.finish(DependentExecute.java:385)
	at org.apache.dolphinscheduler.server.master.engine.executor.plugin.dependent.DependentTaskTracker.isAllDependentTaskFinished(DependentTaskTracker.java:238)
	at org.apache.dolphinscheduler.server.master.engine.executor.plugin.dependent.DependentTaskTracker.getDependentTaskStatus(DependentTaskTracker.java:109)
	at org.apache.dolphinscheduler.server.master.engine.executor.plugin.dependent.DependentLogicTask.getTaskExecutionState(DependentLogicTask.java:76)
	at org.apache.dolphinscheduler.server.master.engine.executor.LogicTaskExecutor.doTrackTaskPluginStatus(LogicTaskExecutor.java:44)
	at org.apache.dolphinscheduler.task.executor.AbstractTaskExecutor.trackTaskExecutorState(AbstractTaskExecutor.java:95)
	at org.apache.dolphinscheduler.task.executor.worker.AbstractTaskExecutorWorker.trackTaskExecutorState(AbstractTaskExecutorWorker.java:38)
	at org.apache.dolphinscheduler.task.executor.worker.TaskExecutorWorker.start(TaskExecutorWorker.java:72)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.sql.SQLSyntaxErrorException: Unknown column 'instance.workflow_instance_id' in 'where clause'
	at com.mysql.cj.jdbc.exceptions.SQLError.createSQLException(SQLError.java:120)
	at com.mysql.cj.jdbc.exceptions.SQLExceptionsMapping.translateException(SQLExceptionsMapping.java:122)
	at com.mysql.cj.jdbc.ClientPreparedStatement.executeInternal(ClientPreparedStatement.java:916)
	at com.mysql.cj.jdbc.ClientPreparedStatement.execute(ClientPreparedStatement.java:354)
	at com.zaxxer.hikari.pool.ProxyPreparedStatement.execute(ProxyPreparedStatement.java:44)
	at com.zaxxer.hikari.pool.HikariProxyPreparedStatement.execute(HikariProxyPreparedStatement.java)
	at org.apache.ibatis.executor.statement.PreparedStatementHandler.query(PreparedStatementHandler.java:64)
	at org.apache.ibatis.executor.statement.RoutingStatementHandler.query(RoutingStatementHandler.java:79)
	at sun.reflect.GeneratedMethodAccessor84.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:64)
	at com.sun.proxy.$Proxy167.query(Unknown Source)
	at org.apache.ibatis.executor.SimpleExecutor.doQuery(SimpleExecutor.java:63)
	at org.apache.ibatis.executor.BaseExecutor.queryFromDatabase(BaseExecutor.java:325)
	at org.apache.ibatis.executor.BaseExecutor.query(BaseExecutor.java:156)
	at com.baomidou.mybatisplus.extension.plugins.MybatisPlusInterceptor.intercept(MybatisPlusInterceptor.java:81)
	at org.apache.ibatis.plugin.Plugin.invoke(Plugin.java:62)
	at com.sun.proxy.$Proxy166.query(Unknown Source)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:151)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:145)
	at org.apache.ibatis.session.defaults.DefaultSqlSession.selectList(DefaultSqlSession.java:140)
	at sun.reflect.GeneratedMethodAccessor81.invoke(Unknown Source)
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.lang.reflect.Method.invoke(Method.java:498)
	at org.mybatis.spring.SqlSessionTemplate$SqlSessionInterceptor.invoke(SqlSessionTemplate.java:427)
	... 36 common frames omitted
```

```
2025-05-15 18:41:44.606 ERROR

org.apache.ibatis.bindinglBindingException: Invalid bound statement (not found) :
org.apache.dolphinscheduler.dao.mapper.TaskInstanceContextMapper.updateTaskInstanceContextByTaskInstanceIdAndContextType
```

### What you expected to happen

running normal.

### How to reproduce

aboved.

### Anything else

_No response_

### Version

3.3.0-alpha

### Are you willing to submit PR?

- [ ] Yes I am willing to submit a PR!

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

--- a/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskInstanceContextMapper.xml
+++ b/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskInstanceContextMapper.xml
@@ -39,7 +39,7 @@
         AND context_type = #{contextType}
     </delete>
 
-    <update id="updateByTaskInstanceId">
+    <update id="updateTaskInstanceContextByTaskInstanceIdAndContextType">
         UPDATE t_ds_task_instance_context
         SET
             context = #{context}

--- a/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskInstanceMapper.xml
+++ b/dolphinscheduler-dao/src/main/resources/org/apache/dolphinscheduler/dao/mapper/TaskInstanceMapper.xml
@@ -245,7 +245,7 @@
         select task_code, max(end_time) as max_end_time
         from t_ds_task_instance
         where 1=1 and test_flag = #{testFlag}
-        and instance.workflow_instance_id = #{workflowInstanceId}
+        and workflow_instance_id = #{workflowInstanceId}
         <if test="taskCodes != null and taskCodes.size() != 0">
             and task_code in
             <foreach collection="taskCodes" index="index" item="i" open="(" separator="," close=")">

